core/runtime/src/main/java/io/quarkus/runtime/configuration/DurationConverter.java

```diff
@@ -0,0 +1,45 @@
+package io.quarkus.runtime.configuration;
+
+import static io.quarkus.runtime.configuration.ConverterSupport.DEFAULT_QUARKUS_CONVERTER_PRIORITY;
+
+import java.time.Duration;
+import java.time.format.DateTimeParseException;
+
+import javax.annotation.Priority;
+
+import org.eclipse.microprofile.config.spi.Converter;
+
+/**
+ * A converter for a {@link Duration} interface.
+ */
+@Priority(DEFAULT_QUARKUS_CONVERTER_PRIORITY)
+public class DurationConverter implements Converter<Duration> {
+    public DurationConverter() {
+    }
+
+    /**
+     * The converter accepts a value which start with a number by implicitly appending `PT` to it.
+     * If the value consists only of a number, it implicitly treats the value as seconds.
+     * Otherwise, tries to convert the value assuming that it is in the accepted ISO-8601 duration format.
+     *
+     * @param value
+     * @return {@link Duration}
+     */
+    @Override
+    public Duration convert(String value) {
+        if (value.matches("[-+]?\\d+")) {
```

Comment:
Please precompile these into `Pattern` objects and store them in `static final` fields.

Reply:
@dmlloyd thanks for the review. Addressing it now.

Reply:
Done.